react/features/remote-video-menu/components/web/MuteEveryoneDialog.js

```diff
@@ -0,0 +1,137 @@
+/* @flow */
+
+import React from 'react';
+
+import {
+    AUDIO_MUTE,
+    createToolbarEvent,
+    sendAnalytics
+} from '../../../analytics';
+import { Dialog } from '../../../base/dialog';
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import { setAudioMuted } from '../../../base/media';
+import UIEvents from '../../../../../service/UI/UIEvents';
+
+import AbstractMuteRemoteParticipantDialog, {
+    type Props as AbstractProps
+} from '../AbstractMuteRemoteParticipantDialog';
+
+declare var APP: Object;
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link MuteEveryoneDialog}.
+ */
+type Props = AbstractProps & {
+
+    /**
+     * The ID of the local participant.
+     */
+    localParticipantId: string,
+
+    /**
+     * The IDs of the remote participants to be muted.
+     */
+    remoteParticipantIds: Array<string>,
+
+     /**
+     * The IDs of the remote participants to exclude from being muted.
+     */
+    exclude: Array<string>,
+};
+
+/**
+ * A React Component with the contents for a dialog that asks for confirmation
+ * from the user before muting a remote participant.
+ *
+ * @extends Component
+ */
+class MuteEveryoneDialog extends AbstractMuteRemoteParticipantDialog<Props> {
+    static defaultProps = {
+        exclude: [],
+        muteLocal: false
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t } = this.props;
+
+        return (
+            <Dialog
+                okKey = 'dialog.muteMultipleParticipantButton'
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.muteMultipleParticipantTitle'
+                width = 'small'>
+                <div>
+                    { t('dialog.muteMultipleParticipantBody') }
+                </div>
+            </Dialog>
+        );
+    }
+
+    _onSubmit: () => boolean;
+
+    /**
+     * Callback to be invoked when the value of this dialog is submitted.
+     *
+     * @returns {boolean}
+     */
+    _onSubmit() {
+        const {
+            dispatch,
+            exclude,
+            localParticipantId,
+            remoteParticipantIds
+        } = this.props;
+
+        remoteParticipantIds
+            .filter(id => !exclude.includes(id))
+            .forEach(id => {
+                if (id === localParticipantId) {
```

Comment:
When you are looping trough the remote participants probably you won't have the local participant id there at all :)

Reply:
The local participant was there while testing..

Reply:
OK. I see. Than please name your props better!

Reply:
OK. Than obviously we have a naming problem here! Please fix this!

Also we have the `.local` prop for every participant.

Reply:
> OK. Than obviously we have a naming problem here! Please fix this!
> 
> Also we have the `.local` prop for every participant.

renamed the prop. not sure what you mean by this

Reply:
I meant that while looping trough the participant you can check if the participant is local or not by checking the `.local` property. This doesn't matter now.